docs/data/material/guides/shadow-dom/shadow-dom.md

```diff
@@ -0,0 +1,64 @@
+# Shadow DOM
+
+<p class="description"><a href="https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_shadow_DOM" target="_blank" rel="noopener nofollow">Shadow DOM</a> can be helpful to isolate part of the app from unwanted global styles.</p>
+
+## Steps
+
+### 1. Styles
+
+In order for styles to work in Shadow DOM, they need to be applied inside of the Shadow DOM:
+
+```tsx
+const container = document.querySelector('#root');
+const shadowContainer = container.attachShadow({ mode: 'open' });
+const emotionRoot = document.createElement('style');
+const shadowRootElement = document.createElement('div');
+shadowContainer.appendChild(emotionRoot);
+shadowContainer.appendChild(shadowRootElement);
+
+const cache = createCache({
+  key: 'css',
+  prepend: true,
+  container: emotionRoot,
+});
+
+ReactDOM.createRoot(shadowRootElement).render(
+  <CacheProvider value={cache}>
+    <App />
+  </CacheProvider>,
+);
+```
+
+### 2. Theme
+
+MUI components like `Menu`, `Dialog`, `Popover` and others use [React Portal](https://reactjs.org/docs/portals.html) to render a new "subtree" in a container outside of current DOM hierarchy.
```

Comment:
> Replacing MUI with Material UI to be more specific.

I would keep MUI, since it's relevant for MUI X and probably for Joy (in the future)

> Should we replace this Portal link with our own Portal docs?

Sure 👍

Reply:
> I would keep MUI, since it's relevant for MUI X and probably for Joy (in the future)

More and more I'm getting hints for an MUI-company-wide documentation 👁️ 👀 🙄 
_(stuff that isn't particular to one or two products but rather a company-wide philosophy/convention/approach)_

Reply:
Huh, I didn't even realise that this guide will be only available in Material UI docs.
I think a lot of pages from "How To Guides" menu section could make sense in other MUI Core products, and sometimes in MUI X as well.
We'll definitely need top-level docs at some point.